JS/Lesson16/goksel/script.js

```diff
@@ -172,17 +195,182 @@ isPositiveNegativeZero(0);
 // 18.5 – 24.9	Healthy Weight
 // 25.0 – 29.9	Overweight
 // 30.0 and Above	Obesity
+let weight = 0;
+let height = 0;
+function calcBodyMassIndex(weight, height){
+	let index = weight / (height ** 2);
+	if(index < 18.5){
+		console.log("Your BMI is " + index.toFixed(2) + " Underweight." );
+	}
+	else if(index > 18.5 && index < 24.9){
+		console.log("Your BMI is " + index.toFixed(2) + " Healthy Weight.");
+	}
+	else if(index > 25 && index < 29.9){
+		console.log("Your BMI is " + index.toFixed(2) + " Overweight." );
+	}
+	else if(index >= 30){
+		console.log("Your BMI is " + index.toFixed(2) + " Obesity.")
+	}
+	else{
+		console.log("Your weigt and height should be positive value");
+	}
+}
+
+calcBodyMassIndex(50,1.85);
+calcBodyMassIndex(60,1.7);
+calcBodyMassIndex(70,1.6);
+calcBodyMassIndex(120,1.5);
 
 // 9. Write a program that takes in a person's age and checks if they are a child (age 0-12), teenager (age 13-19), adult (age 20-64), or senior citizen (age 65 and above) using if else statement.
 
+function checkYourAge(age){
+	if(age > 0 && age <= 12){
+		console.log("You are " + age + " years old and you are a \"child\".");
+	}
+	else if(age > 13 && age <= 19){
+		console.log("You are " + age + " years old and you are a \"teenager\".");
+	}
+	else if(age > 20 && age <= 64){
+		console.log("You are " + age + " years old and you are an \"adult\".");
+	}
+	else if(age >= 65){
+		console.log("You are " + age + " years old and you are a \"senior citizen\".");
+	}
+	else {
+		console.log("You entered a wring value! Your age should be a positive value.")
+	}
+}
+
+checkYourAge(7);
+checkYourAge(12);
+checkYourAge(17);
+checkYourAge(32);
+checkYourAge(70);
+checkYourAge(-3);
+
+
 // 10. Write a program that takes age as a number and whether they are a student (yes or no). If the age is less than 18 and the user is a student, print "You are a student and not old enough to vote". If the age is between 18 and 65 (inclusive) and the user is not a student, print "You are eligible to vote". If the age is greater than 65, print "You are a senior citizen". If the user is a student but their age is greater than or equal to 18, print "You are old enough to vote".
 
+
+
+
+function isStudentAllowedToVote(age,student){
```

Comment:
In programming, when you have a variable that can only mean something like 'yes' or 'no', it's common to use a boolean type value instead (true / false). Make the student parameter in this function a boolean

Reply:
function isStudentAllowedToVote(age,student){
	if(age < 18 && student === true){
		console.log("You are a student and not old enough to vote");
	}
	else if(age >= 18 && age <= 65 && student === false){
		console.log("You are eligible to vote");
	}
	else if(age > 65){
		console.log("You are a senior citizen");
	}
	else if(age >= 18 && student === true){
		console.log("You are old enough to vote");
	}
	else{
		console.log("You entered a wrong value! Your age should be greater than 0");
	}
}
isStudentAllowedToVote(17,true);
isStudentAllowedToVote(24,false);
isStudentAllowedToVote(70);
isStudentAllowedToVote(24, true);
isStudentAllowedToVote(-4);

Reply:
Looks good but please change the code accordingly!